spdx-utils/src/main/kotlin/model/SpdxFile.kt

```diff
@@ -0,0 +1,193 @@
+     */
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    val licenseComments: String = "",
+
+    /**
+     * The concluded license for the file as SPDX expression. To represent a not present value [SpdxConstants.NONE] or
+     * [SpdxConstants.NOASSERTION] must be used.
+     */
+    val licenseConcluded: String,
+
+    /**
+     * The license information found in this file. To represent a not present value [SpdxConstants.NONE] or
+     * [SpdxConstants.NOASSERTION] must be used.
+     */
+    val licenseInfoInFiles: List<String>,
+
+    /**
+     * License notices or other such related notices found in the file. This may include copyright statements.
+     */
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    val noticeText: String = ""
+) {
+    enum class Type {
+        /**
+         * The file is associated with a specific application type (MIME type of application e.g. .exe).
+         */
+        APPLICATION,
+
+        /**
+         * Indicates the file is an archive file.
+         */
+        ARCHIVE,
+
+        /**
+         * The file is associated with an audio file (MIME type of audio, e.g. .mp3).
+         */
+        AUDIO,
+
+        /**
+         * Indicates the file is not a text file.
+         */
+        BINARY,
+
+        /**
+         * The file serves as documentation.
+         */
+        DOCUMENTATION,
+
+        /**
+         * The file is associated with an picture image file (MIME type of image, e.g. .jpg, .gif).
+         */
+        IMAGE,
+
+        /**copyrightText
+         * Indicates the file is not a source, archive or binary file.
+         */
+        OTHER,
+
+        /**
+         * Indicates the file is a source code file.
+         */
+        SOURCE,
+
+        /**
+         * The file is an SPDX document.
+         */
+        SPDX,
+
+        /**
+         * The file is a human readable text file (MIME type of text).
+         */
+        TEXT,
+
+        /**
+         * The file is associated with a video file (MIME type of video, e.g. .avi, .mkv, .mp4)
+         */
+        VIDEO;
+    }
+
+    init {
+        require(checksums.filter { it.algorithm == SpdxChecksum.Algorithm.SHA1 }.isNotEmpty()) {
+            "At least one SHA1 checksum must be provided."
+        }
+
+        require(copyrightText.isNotBlank()) {
+            "The copyright text must not be blank."
+        }
+
+        require(filename.isNotBlank()) { "The filename must not be blank." }
+
+        require(licenseConcluded.isSpdxExpressionOrNotPresent()) {
+            "The license concluded must be either an SpdxExpression, 'NONE' or 'NOASSERTION', but was " +
+                "$licenseConcluded."
+        }
+
+        // TODO: The check for [licenseInfoInFiles] can be made more strict, but the SPDX specification is not exact
+        // enough yet to do this safely.
+        licenseInfoInFiles.filterNot { it.isSpdxExpressionOrNotPresent() }.let {
+            require(it.isEmpty()) {
```

Comment:
`require(licenseInfoInFiles.all { it.isSpdxExpressionOrNotPresent() })`

Reply:
Doing so would either introduce redundant code for the error message or make the error message not mentioned the wrong elements. I prefer to keep it as is.